overlord/configstate/configstate.go

```diff
@@ -65,7 +65,7 @@ func canConfigure(st *state.State, snapName string) error {
 	}
 
 	// the "snapd" snap cannot be configured yet
-	if snapName == "snapd" {
+	if snapst.SnapType == string(snap.TypeSnapd) {
```

Comment:
this should be snapst.Type() == snap.TypeSnapd,  unless I'm missing something

Reply:
Thank you, fixed.